packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -124,15 +126,36 @@ export function createOdspNetworkError(
     props: ITelemetryProperties = {},
 ): LoggingError & OdspError & IFacetCodes {
     let error: LoggingError & OdspError & IFacetCodes;
+    const parseResult = tryParseErrorResponse(responseText);
+    let facetCodes: string[] | undefined;
+    let innerMostErrorCode: string | undefined;
+    if (parseResult.success) {
+        // Log the whole response if it looks like the error format we expect
+        props.response = responseText;
+        const errorResponse = parseResult.errorResponse;
+        facetCodes = parseFacetCodes(errorResponse);
+        if (facetCodes !== undefined) {
+            innerMostErrorCode = facetCodes[0];
+            props.innerMostErrorCode = innerMostErrorCode;
+        }
+    }
     switch (statusCode) {
         case 400:
             error = new GenericNetworkError(errorMessage, false, { statusCode });
             break;
         case 401:
         case 403:
-            const claims = response?.headers ? parseAuthErrorClaims(response.headers) : undefined;
-            const tenantId = response?.headers ? parseAuthErrorTenant(response.headers) : undefined;
-            error = new AuthorizationError(errorMessage, claims, tenantId, { statusCode });
+            if (innerMostErrorCode === OdspErrorType.serviceReadOnly) {
+                error = new RetryableError(
+                    errorMessage,
+                    OdspErrorType.serviceReadOnly,
+                    { statusCode: serviceReadOnlyErrorCode },
```

Comment:
I do not see a need to define status code and add it here. My preference would be to not have status codes at all (with driver doing projection to driver errors), but it's useful thing for logging where real code is returned by service (to validate that we mapped it correctly). Here - there is no code from service, so no need to invent one.